antenna/breakpad_resource.py

```diff
@@ -293,7 +293,12 @@ def extract_payload(self, req):
         request_env = dict(req.env)
         request_env["QUERY_STRING"] = ""
 
-        fs = cgi.FieldStorage(fp=data, environ=request_env, keep_blank_values=1)
+        try:
+            fs = cgi.FieldStorage(fp=data, environ=request_env, keep_blank_values=1)
+        except ValueError:
+            # cgi.FieldStorage throws a ValueError if the boundary is not a str;
+            # treat this as an invalid payload
+            raise MalformedCrashReport("malformed_boundary")
```

Comment:
The existing testlib makes this a really hard test to write. I'm tinkering with one.